portfolio/src/main/webapp/script.js

```diff
@@ -26,3 +26,43 @@ function addRandomGreeting() {
   const greetingContainer = document.getElementById('greeting-container');
   greetingContainer.innerText = greeting;
 }
+var slideIndex = 1;
+showSlides(slideIndex);
+
+function plusSlides(offset) {
+  showSlides(slideIndex += offset);
+}
+
+function currentSlide(newSlideIndex) {
+  showSlides(slideIndex = newSlideIndex);
+}
+
+function showSlides(activatedSlide) {
+  var i;
+  var slides = document.getElementsByClassName("slides");
+  var dots = document.getElementsByClassName("dot");
+
+  if (activatedSlide > slides.length) {
+    slideIndex = 1;
+  }  
+  if (activatedSlide < 1) {
+    slideIndex = slides.length
+  }
+
+  // Make all photos invisible
```

Comment:
Instead of adding a comment around the for loop for explanation, you could factor out a function

function hideAllSlides() {
 ...
}

function disableAllDots() {
 ...
}

Then the code will explain itself and you can save yourself from requiring any additional comments.